.pre-commit-config.yaml

```diff
@@ -0,0 +1,36 @@
+default_language_version:
```

Comment:
Adding this pre-commit config like this is a strong No from me. It requires we maintain settings in multiple places. We have configurations for these already, we have tests that will fail if someone didn't comply, and we have an easy way to run anything which should be automated. IDEs will format with black using our settings already just by setting black as the formatting provider. pre-commit runs in an isolated environment each time, ignoring our central settings. 

As an example of the trouble this can cause, this has args which mismatch the ones we are using for one of the tools here, and you just created it. The fewer places we duplicate settings, the better.

This just isn't suited to the project.

Reply:
Are we sure thats the case when I was running the pre-commit hook localy based on this file it was using the settings from the setup.cfg, the individual files and the .ini file 
Reason I added here was something user could easily run to process all the formatting  in 1 command, and ensure everything will pass the Style check

Reply:
`make reformat` already handles the style bit. And yes, pre-commit creates a virtualenv specifically for all of this, using the versions specified here, not in setup.cfg and using the args specified here over the ones in our configuration files, which makes this another place we would need to update multiple things rather than just let people be responsible for their own development environments and catch it at the tests.